examples/computer_vision/steps/evaluate.py

```diff
@@ -0,0 +1,86 @@
+# Apache Software License 2.0
+#
+# Copyright (c) ZenML GmbH 2025. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Model evaluation step using YOLO validation."""
+
+from pathlib import Path
+from typing import Annotated, Dict, Union
+
+from ultralytics import YOLO
+
+from zenml import step
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+@step
+def evaluate_model(
+    trained_model: YOLO,
+    dataset_path: Path,
+) -> Annotated[Dict[str, Union[float, str]], "metrics"]:
+    """Evaluate the trained YOLO model on the validation dataset.
+
+    This step runs validation on the trained model to get performance metrics
+    like mAP (mean Average Precision), precision, and recall.
+
+    Args:
+        trained_model: The trained YOLO model
+        dataset_path: Path to the dataset directory with YOLO format
+
+    Returns:
+        Dictionary containing validation metrics (mAP50, mAP50-95, precision, recall)
+    """
+    logger.info("Running model validation...")
+
+    # Find the data.yaml file
+    data_yaml = dataset_path / "dataset.yaml"
+
+    if not data_yaml.exists():
+        logger.warning(f"Dataset config not found at {data_yaml}")
```

Comment:
Somehow wonder if it's not better to error out completely vs giving these null
results? I know all the models love to try/except everything so nothing ever
fails, but it feels like you'd want it to error out here? (unless this is caught
somewhere else?)